client/src/components/app.jsx

```diff
@@ -19,15 +33,67 @@ class App extends React.Component {
   }
 
   componentDidMount() {
-    this.fetchHome(val => {
+    // reference to specific home listing
+    var param = window.location.href.slice(27, -1);
+    this.fetchHome(param, val => {
```

Comment:
I see you're using a promise with a callback argument. Although not important yet, it's better to do all your work in a promise/then() function, so that in case of errors the catch() will be called.

Reply:
Refactored to avoid using callbacks.